base/util_testing.go

```diff
@@ -235,56 +248,115 @@ func (tbm *TestBucketManager) OpenTestBucket() (bucketExists bool, err error) {
 
 	IncrNumOpenBuckets(tbm.BucketSpec.BucketName)
 
-	cluster, err := gocb.Connect(tbm.BucketSpec.Server)
+	tbm.Bucket, err = GetCouchbaseBucketGoCB(tbm.BucketSpec)
 	if err != nil {
 		return false, err
 	}
-	tbm.Cluster = cluster
-
-	tbm.ClusterManager = cluster.Manager(tbm.AdministratorUsername, tbm.AdministratorPassword)
-
-	username, password, _ := tbm.BucketSpec.Auth.GetCredentials()
-	bucket, err := tbm.Cluster.OpenBucket(tbm.BucketSpec.BucketName, password)
-	if err != nil {
-		// Authentication failure should return an explicit error as we can't continue from here.
-		if pkgerrors.Cause(err) == gocb.ErrAuthError {
-			log.Printf("Unable to authenticate as %s: %v", username, err)
-			return false, err
-		}
-
-		// There could be other errors here, but we assume the bucket doesn't exist, and may be able to continue.
-		// TODO: should check returned error type
-		log.Printf("GoCB error opening bucket: %v", err)
-		return false, nil
-	}
-	tbm.Bucket = bucket
-
-	tbm.BucketManager = tbm.Bucket.Manager(username, password)
 
 	return true, nil
 
 }
 
-func (tbm *TestBucketManager) Close() error {
-	return tbm.Bucket.Close()
+func (tbm *TestBucketManager) Close() {
+	tbm.Bucket.Close()
 }
 
 // GOCB doesn't currently offer a way to do this, and so this is a workaround to go directly
 // to Couchbase Server REST API.
 // See https://forums.couchbase.com/t/is-there-a-way-to-get-the-number-of-items-in-a-bucket/12816/4
 // for GOCB discussion.
 func (tbm *TestBucketManager) BucketItemCount() (itemCount int, err error) {
-	return GoCBBucketItemCount(tbm.Bucket, tbm.BucketSpec, tbm.AdministratorUsername, tbm.AdministratorPassword)
+	return GoCBBucketItemCount(tbm.Bucket.Bucket, tbm.BucketSpec, tbm.AdministratorUsername, tbm.AdministratorPassword)
+}
+
+func (tbm *TestBucketManager) DropIndexes() error {
+	return DropAllBucketIndexes(tbm.Bucket)
+}
+
+// Reset bucket state
+func DropAllBucketIndexes(gocbBucket *CouchbaseBucketGoCB) error {
+
+	// Retrieve all indexes
+	indexes, err := getIndexes(gocbBucket)
+	if err != nil {
+		return err
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(indexes))
+
+	asyncErrors := make(chan error, len(indexes))
+	defer close(asyncErrors)
+
+	for _, index := range indexes {
+
+		go func(indexToDrop string) {
+
+			defer wg.Done()
+
+			log.Printf("Dropping index %s...", indexToDrop)
+			dropErr := gocbBucket.DropIndex(indexToDrop)
+			if dropErr != nil {
+				asyncErrors <- dropErr
+			}
+			log.Printf("...successfully dropped index %s", indexToDrop)
+		}(index)
+
+	}
+
+	// Wait until all goroutines finish
+	wg.Wait()
+
+	// Check if any errors were put into the asyncErrors channel.  If any, just return the first one
+	select {
+	case asyncError := <-asyncErrors:
+		return asyncError
+	default:
+	}
+
+	return nil
+}
+
+// Get a list of all index names in the bucket
```

Comment:
Might be a good time to replace this w/ gocb's BucketManager.GetIndexes().

Reply:
Thanks for catching. Fixed